### PR TITLE
Add about dialog

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
About window added.
Tcl command show_about added. 
Include into the tests.

image

### CODE REVIEW

**Comment by ravic-rs** [comment]
Verify command line is same as GUI about.
https://github.com/RapidSilicon/Raptor/issues/129#issuecomment-1175505960

File: src/Main/AboutWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#include <QUrl>
+
+namespace FOEDAG {
+
+AboutWidget::AboutWidget(const ProjectInfo &info, QWidget *parent)
+    : QDialog{parent} {
+  QLabel *label = new QLabel(this);
+  QPushButton *close = new QPushButton("Close", this);
+
+#ifdef PRODUCTION_BUILD
```
[the remark this answers is not in the document]
Comment (KochynVolodymyr) [reply src/Main/AboutWidget.cpp]: Fixed

File: CMakeLists.txt [hunk 2]
```diff
@@ -67,6 +67,13 @@ endif()
 # NOTE: Policy changes has to happen before adding any subprojects
 cmake_policy(SET CMP0091 NEW)
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+OPTION(PRODUCTION_BUILD_OPT "Define Production or Engineering build" OFF)
```
Comment (alain-rs) [inline CMakeLists.txt:70]: I did check in the official PRODUCTION_BUILD flags yesterday. 
Don't use the flags above, use the ones used in the whole source code.

Comment (KochynVolodymyr) [reply CMakeLists.txt]: Done

**Comment by alain-rs** [comment]
@KochynVolodymyr, I tried to merge the conflicts in the GUI, but my changes did not stick.
You have to resolve the conflict manually.
Beware the new code from @aram-rs is needed.

**Comment by volodymyrkochyn** [comment]
@alain-rs Done. rebase is doing all magic

**Comment by alain-rs** [comment]
@KochynVolodymyr, can you rerun the valgrind tests locally?
make test/valgrind

**Comment by KochynVolodymyr** [comment]
@alain-rs I already did it few times. Locally I don't have this problem.

**Comment by alain-rs** [comment]
@KochynVolodymyr 
That test is broken, it is a negative test, check the Makefile. It has to do with the change in bahavior of add_design_file 


```
alain@alain-xps:~/FOEDAG$ make test/valgrind 
cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_RULE_MESSAGES=on -DPRODUCTION_BUILD= -DNO_TCMALLOC=On  -S . -B dbuild
CMAKE_SYSTEM_NAME: Linux
LINUX MODE
-- GIT_HASH = 0e26e57
-- Configuring done
-- Generating done
-- Build files have been written to: /home/alain/FOEDAG/dbuild
valgrind --tool=memcheck --log-file=valgrind.log dbuild/bin/foedag --batch --script tests/TestBatch/hello.tcl ; 
Hello!
grep "ERROR SUMMARY: 0" valgrind.log
==652975== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/foedag --replay tests/TestGui/gui_start_stop.tcl;
TEST_LOOP_ENTERED
GUI START
GUI STOP
GUI START
GUI STOP
GUI EXIT
TEST_LOOP_EXITED
Tcl Exit
grep "ERROR SUMMARY: 0" valgrind_gui.log 
==653055== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/newproject --replay tests/TestGui/gui_new_project.tcl
TEST_LOOP_ENTERED
GUI START
NEXT
NEXT
NEXT
NEXT
NEXT
GUI STOP
GUI EXIT
TEST_LOOP_EXITED
Tcl Exit
grep "ERROR SUMMARY: 0" valgrind_gui.log
==653180== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/projnavigator --replay tests/TestGui/gui_project_navigator.tcl
TEST_LOOP_ENTERED
GUI START
GUI STOP
GUI START
GUI STOP
GUI EXIT
TEST_LOOP_EXITED
Tcl Exit
grep "ERROR SUMMARY: 0" valgrind_gui.log
==653258== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/texteditor --replay tests/TestGui/gui_text_editor.tcl
TEST_LOOP_ENTERED
TEXT EDITOR GUI START
TEXT EDITOR GUI OPENFILE
QImage::scaled: Image is a null image
TEXT EDITOR GUI STOP
GUI EXIT
TEST_LOOP_EXITED
Tcl Exit
grep "ERROR SUMMARY: 0" valgrind_gui.log
==653290== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/newfile --replay tests/TestGui/gui_new_file.tcl
TEST_LOOP_ENTERED
NEWFILE GUI START
NEWFILE GUI STOP
NEWFILE GUI START
NEWFILE GUI STOP
GUI EXIT
TEST_LOOP_EXITED
Tcl Exit
grep "ERROR SUMMARY: 0" valgrind_gui.log
==653321== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/console_test --replay tests/TestGui/gui_console.tcl
SUCCESS
SUCCESS
SUCCESS
SUCCESS
SUCCESS
grep "ERROR SUMMARY: 0" valgrind_gui.log
==653392== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/foedag --replay tests/TestGui/gui_foedag.tcl
TEST_LOOP_ENTERED
GUI START
GUI STOP
GUI START
TEXT EDITOR GUI OPENFILE
TEXT EDITOR GUI OPENFILE
NEW PROJECT START
NEXT
NEXT
NEXT
NEXT
NEXT
NEW PROJECT STOP
GUI STOP
add_design_file TEST
PASSED: Caught negative test
File(s) not exist: /home/alain/FOEDAG/asd.v 

grep "ERROR SUMMARY: 0" valgrind_gui.log
make: *** [Makefile:104: test/valgrind] Error 1
```

**Comment by alain-rs** [comment]
Yes, the problem is real, this is my local run.

the file does not exist, so return error.

**Comment by ravic-rs** [comment]
> check the latest run. It fails in a different place

Does github store history of all valgrind tests run on this PR ?

**Comment by KochynVolodymyr** [comment]
> Does github store history of all valgrind tests run on this PR ?

Yes.

**Comment by ravic-rs** [comment]
> > Does github store history of all valgrind tests run on this PR ?
> 
> Yes.

Ok, I found only runs from today. I was looking for logs when this PR was opened.
image

**Comment by alain-rs** [comment]
Look at the test:
```

puts "add_design_file TEST";flush stdout;source tests/TestGui/foedag_proc.tcl;add_des_file
alain@alain-xps:~/FOEDAG$ cat tests/TestGui/foedag_proc.tcl
#Copyright 2021 The Foedag team

#GPL License

#Copyright (c) 2021 The Open-Source FPGA Foundation

#This program is free software: you can redistribute it and/or modify
#it under the terms of the GNU General Public License as published by
#the Free Software Foundation, either version 3 of the License, or
#(at your option) any later version.

#This program is distributed in the hope that it will be useful,
#but WITHOUT ANY WARRANTY; without even the implied warranty of
#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#GNU General Public License for more details.

#You should have received a copy of the GNU General Public License
#along with this program.  If not, see <http://www.gnu.org/licenses/>.

proc add_des_file {} { 
gui_start
create_design asd
if { [catch {add_design_file asd.v} result] } {
    puts "PASSED: Caught negative test"
    puts $result
    exit 0
}
error "add_design_file expect failed"
exit 1
}
```
I think the add_design_file behavior changed and the test was not updated?

**Comment by KochynVolodymyr** [comment]
I see fail here
xvfb-run --auto-servernum --server-args="-screen 0, 1280x1024x24" valgrind --tool=memcheck --log-file=valgrind_gui.log dbuild/bin/newfile --replay tests/TestGui/gui_new_file.tcl
TEST_LOOP_ENTERED
NEWFILE GUI START
NEWFILE GUI STOP
NEWFILE GUI START
NEWFILE GUI STOP
GUI EXIT
TEST_LOOP_EXITED
Tcl Exit
grep "ERROR SUMMARY: 0" valgrind_gui.log
make: *** [Makefile:98: test/valgrind] Error 1
Error: Process completed with exit code 2.

**Comment by alain-rs** [comment]
The test is not correct:

In the Makefile, a negative test should be like:

	$(XVFB) ./dbuild/bin/console_test --replay tests/TestGui/gui_console_negative_test.tcl && exit 1 || (echo "PASSED: Caught negative test")

You added a sub test to an existing test without using the right Makefile construct for negative test.
Create a different test to test add_design_file that is a negative test (One negative test per test). Don't mix and match with a valgrind test for that.

**Comment by alain-rs** [comment]
BTW, valgrind tests should never fail, good thing we caught this before it gets merged.

**Pull request merged**